### DISCUSSION TITLE
Is beam search ready?

### DISCUSSION THREAD

**Discussion by Sunt-ing** [ggml-org/llama.cpp#7224] [Q&A] [answered]
I see efforts to implement beam search in llama.cpp and find relevant code, but I cannot find documents about how to set the relevant parameters (e.g., num_beams). Is this feature ready?

**Comment by ggerganov** [comment, marked as answer]
There was an initial implementation (still exposed via `llama.h`), but IMO it has to be reimplemented from scratch as a standalone example

**Reply by laelhalawani** [reply]
why is it wrong or is the logic correct?
